client/src/modules/dashboard/Chapters/pages/ChapterPage.tsx

```diff
@@ -1,21 +1,37 @@
-import { Heading, Link, Box, HStack } from '@chakra-ui/layout';
+import { Heading, Link, Box, HStack, Flex, Text } from '@chakra-ui/layout';
+import { Tabs, TabList, TabPanels, Tab, TabPanel } from '@chakra-ui/tabs';
+
 import { LinkButton } from 'chakra-next-link';
 import { NextPage } from 'next';
-import React from 'react';
+import React, { useRef } from 'react';
+import { Button, useDisclosure, AlertDialog } from '@chakra-ui/react';
 import { Card } from '../../../../components/Card';
+import { SettingAlertDialog } from '../../../../components/SettingAlert';
 import ProgressCardContent from '../../../../components/ProgressCardContent';
-import { useChapterQuery } from '../../../../generated/graphql';
+import {
+  useChapterQuery,
+  useDeleteChapterMutation,
+} from '../../../../generated/graphql';
 import { useParam } from '../../../../hooks/useParam';
 import styles from '../../../../styles/Page.module.css';
 import { Layout } from '../../shared/components/Layout';
 
 export const ChapterPage: NextPage = () => {
   const { param: chapterId } = useParam('id');
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const alertDialogFocusElement = useRef(null);
+
+  const [deleteChapter] = useDeleteChapterMutation();
 
   const { loading, error, data } = useChapterQuery({
     variables: { chapterId },
   });
 
+  const clickDelete = () => {
+    deleteChapter({ variables: { chapterId } });
+    window.location.href = '/dashboard/chapters';
```

Comment:
Is there a better way to do this with react?

Reply:
See https://nextjs.org/docs/api-reference/next/router#routerpush

Reply:
Thank you for the link, I couldn't follow the steps in https://nextjs.org/docs/api-reference/next/router#resetting-state-after-navigation well. 

And I don't want to use `useEffect` which I am not comfortable with yet, I am delaying this to post MVP. 

Here is my attempt, if I did something stupid, and it can work with small change. Then please guide me https://github.com/freeCodeCamp/chapter/pull/1237/commits/50f523de8ace70689e87a0e7175417ac5eec71e7

Reply:
The update can be done by passing to the `useDeleteChapterMutation()`, the `refetchQueries` argument with query that should be updated  
```typescript
import { CHAPTERS } from '../../../chapters/graphql/queries';
/* ... */
    const [deleteChapter] = useDeleteChapterMutation({
    refetchQueries: [{ query: CHAPTERS }],
  });
```